homeassistant/components/recorder/purge.py

```diff
@@ -104,6 +108,24 @@ def _purge_state_ids(session: Session, state_ids: list[int]) -> None:
     _LOGGER.debug("Deleted %s states", deleted_rows)
 
 
+def _evict_purged_states_from_old_states_cache(
+    instance: Recorder, purged_state_ids: list[int]
+) -> None:
+    """Evict purged states from the old states cache."""
+    # Make a map from old_state_id to entity_id
+    old_states = instance._old_states  # pylint: disable=protected-access
+    old_state_reversed = {
+        old_state.state_id: entity_id
+        for entity_id, old_state in old_states.items()
+        if old_state.state_id
+    }
+
+    # Evict any purged state from the old states cache
+    old_state_ids = set(old_state_reversed.keys())
```

Comment:
`keys()` is default iterator of a dict. 
```suggestion
    old_state_ids = set(old_state_reversed)
```

Reply:
I think this is going to come in as a `set` in the next turn per https://github.com/home-assistant/core/pull/56877/files#r720064652